win32/src/uuid_win32.c

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "rpc.h"
+
+#include "macro_utils/macro_utils.h"
+
+#include "c_logging/xlogging.h"
+
+#include "c_pal/uuid.h"
+
+static int is_UUID_T_and_UUID_same_size[sizeof(UUID_T) == sizeof(UUID)]; /*just a sanity check*/
+
+/*these are interface requirements*/
+/*Codes_SRS_UUID_02_001: [ If destination is NULL then uuid_produce shall fail and return a non-NULL value. ]*/
+/*Codes_SRS_UUID_02_002: [ uuid_produce shall generate in destination the representation of a UUID (as per RFC 4122). ]*/
+/*Codes_SRS_UUID_02_004: [ uuid_produce shall succeed and return 0. ]*/
+
+static void GUID_to_UUID_T(const GUID* guid, UUID_T uuid)
+{
+    uuid[0] = (guid->Data1 >> 24) & 0xFF;
+    uuid[1] = (guid->Data1 >> 16) & 0xFF;
+    uuid[2] = (guid->Data1 >> 8) & 0xFF;
+    uuid[3] = (guid->Data1) & 0xFF;
+    uuid[4] = (guid->Data2 >> 8) & 0xFF;
+    uuid[5] = (guid->Data2) & 0xFF;
+    uuid[6] = (guid->Data3 >> 8) & 0xFF;
+    uuid[7] = (guid->Data3) & 0xFF;
+    uuid[8] = guid->Data4[0];
+    uuid[9] = guid->Data4[1];
+    uuid[10] = guid->Data4[2];
+    uuid[11] = guid->Data4[3];
+    uuid[12] = guid->Data4[4];
+    uuid[13] = guid->Data4[5];
+    uuid[14] = guid->Data4[6];
+    uuid[15] = guid->Data4[7];
+}
+
+int uuid_produce(UUID_T destination)
+{
+    int result;
+    /*Codes_SRS_UUID_WIN32_02_001: [ If destination is NULL then uuid_produce shall fail and return a non-NULL value. ]*/
+    if (destination == NULL)
+    {
+        LogError("invalid argument UUID_T destination=%p", destination);
+        result = MU_FAILURE;
+    }
+    else
+    {
+        UUID u;
+        /*Codes_SRS_UUID_WIN32_02_002: [ uuid_produce shall call UuidCreate to generate a UUID. ]*/
+        RPC_STATUS rpc_status = UuidCreate(&u);

```

Comment:
>UuidCreate [](http://example.com/codeflow?start=32&length=10)

I'm curious why this doesn't just call `UniqueId_Generate`, which calls `UuidCreate`. Is the intention to remove uniqueid.h entirely? (Sorry maybe I just don't have enough context on this PR)

Reply:
We're getting rid of the UniqueId_xxx, because they generate stuff directly as string. And then in some cases we convert it back from string, which is highly suboptimal.

Reply:
and the plan is to completely remove traces of the "old" uniqueId module. Removal comes as soon as everybody moved on to the new one.

Reply:
Got it, this is the context I was missing from the "have unique UUID" thanks